Packs/BitwardenPasswordManager/Integrations/BitwardenPasswordManagerEventCollector/BitwardenPasswordManagerEventCollector.py

```diff
@@ -0,0 +1,280 @@
+        set_integration_context(context={
+            'token': token,
+            'expires': str(expire_date)
+        })
+
+    def get_events(self, start_date: str = "", end_date: str = "", continuation_token: str = "") -> dict:
+        params = {
+            'start': start_date,
+            'end': end_date,
+            'continuationToken': continuation_token
+        }
+
+        headers = {
+            'Authorization': f'Bearer {self.token}'
+        }
+
+        res = self._http_request(
+            method='GET',
+            url_suffix='/public/events',
+            headers=headers,
+            params=params
+        )
+
+        return res
+
+
+def test_module(client: Client) -> str:
+    try:
+        retrieve_events, last_run = fetch_events(client, max_fetch=1)
+        demisto.log(f"{retrieve_events=}")
+        demisto.log(f"{last_run=}")
+    except DemistoException as e:
+        raise e
+
+    return 'ok'
+
+
+def convert_to_timestamp(date: datetime | None) -> int:
+    """Converts datetime to timestamp"""
+    if date:
+        if isinstance(date, datetime):
+            return int(date.timestamp())
+        elif isinstance(date, int):
+            return int(date)
+    return 0
+
+
+def calculate_fetch_dates(start_date: str, last_run: dict, end_date: str = "") -> tuple:
+    """
+    Calculates the start and end dates for fetching events.
+
+    This function takes the start date and end date provided as arguments.
+    If these are not provided, it uses the last run information to calculate the start and end dates.
+    If the last run information is also not available,
+     it uses the current time as the end date and the time one minute before the current time as the start date.
+
+    Args:
+        start_date (str): The start date for fetching events in '%Y-%m-%dT%H:%M:%SZ' format.
+        last_run_key (str): The key to retrieve the last fetch date from the last run dictionary.
+        last_run (dict): A dictionary containing information about the last run.
+        end_date (str, optional): The end date for fetching events in '%Y-%m-%dT%H:%M:%SZ' format. Defaults to "".
+
+    Returns:
+        tuple: A tuple containing two elements:
+            - The start date as a string in the format '%Y-%m-%dT%H:%M:%SZ'.
+            - The end date as a string in the format '%Y-%m-%dT%H:%M:%SZ'.
+    """
+    now_utc_time = get_current_time()
+    # argument > last run > current time
+    start_date = start_date or last_run.get('last_fetch') or (
+        (now_utc_time - timedelta(minutes=1)).strftime(DATE_FORMAT))
+    # argument > current time
+    end_date = end_date or now_utc_time.strftime(DATE_FORMAT)
+    return start_date, end_date
+
+
+def fetch_events(client: Client, max_fetch: int, start_date_str: str = "", end_date_str: str = "") -> tuple:
+    last_run = demisto.getLastRun()
+    continuation_token = last_run.get("continuationToken", "")
+    demisto.log(f"{continuation_token=}")  # TODO: Remove
```

Comment:
FYI better to use demisto.debug (or demisto.info if we want it to always appear in logs)
demisto.log creates a visible log entry, an old feature that we no longer really use